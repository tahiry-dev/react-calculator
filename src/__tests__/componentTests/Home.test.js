import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Home from '../../components/Home';

describe('Home', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Test Heading for home component', () => {
  it('returns welocome message', () => {
    render(<Home />);
    expect(screen.getByRole('heading')).toHaveTextContent('Welcome to our Page!');
  });
});
