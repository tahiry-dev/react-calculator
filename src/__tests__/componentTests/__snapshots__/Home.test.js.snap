// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders correctly 1`] = `
<div
  className="container"
>
  <h2>
    Welcome to our page!
  </h2>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin fringilla nulla sed commodo efficitur. Curabitur porttitor hendrerit nisl, vitae dapibus nisi imperdiet auctor. Suspendisse a orci dolor. Aliquam pharetra id ligula sit amet egestas. Maecenas tincidunt nulla turpis, eu molestie enim accumsan sit amet. In ut erat suscipit, ornare lectus quis, ullamcorper erat. Pellentesque tempus arcu ut molestie tincidunt. Ut ac leo pretium, vestibulum arcu sit amet, finibus quam. Mauris sit amet tincidunt urna. Phasellus dapibus augue vitae eleifend rhoncus.
  </p>
  <br />
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin fringilla nulla sed commodo efficitur. Curabitur porttitor hendrerit nisl, vitae dapibus nisi imperdiet auctor. Suspendisse a orci dolor. Aliquam pharetra id ligula sit amet egestas. Maecenas tincidunt nulla turpis, eu molestie enim accumsan sit amet. In ut erat suscipit, ornare lectus quis, ullamcorper erat. Pellentesque tempus arcu ut molestie tincidunt. Ut ac leo pretium, vestibulum arcu sit amet, finibus quam. Mauris sit amet tincidunt urna. Phasellus dapibus augue vitae eleifend rhoncus.
  </p>
</div>
`;

exports[`Home renders correctly 1`] = `
<div
  className="container"
>
  <h2>
    Welcome to our Page!
  </h2>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin fringilla nulla sed commodo efficitur. Curabitur porttitor hendrerit nisl, vitae dapibus nisi imperdiet auctor. Suspendisse a orci dolor. Aliquam pharetra id ligula sit amet egestas. Maecenas tincidunt nulla turpis, eu molestie enim accumsan sit amet. In ut erat suscipit, ornare lectus quis, ullamcorper erat. Pellentesque tempus arcu ut molestie tincidunt. Ut ac leo pretium, vestibulum arcu sit amet, finibus quam. Mauris sit amet tincidunt urna. Phasellus dapibus augue vitae eleifend rhoncus.
  </p>
  <br />
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin fringilla nulla sed commodo efficitur. Curabitur porttitor hendrerit nisl, vitae dapibus nisi imperdiet auctor. Suspendisse a orci dolor. Aliquam pharetra id ligula sit amet egestas. Maecenas tincidunt nulla turpis, eu molestie enim accumsan sit amet. In ut erat suscipit, ornare lectus quis, ullamcorper erat. Pellentesque tempus arcu ut molestie tincidunt. Ut ac leo pretium, vestibulum arcu sit amet, finibus quam. Mauris sit amet tincidunt urna. Phasellus dapibus augue vitae eleifend rhoncus.
  </p>
</div>
`;
